server/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStats.java

```diff
@@ -404,6 +433,14 @@ public void add(CommonStats stats) {
                 shards = shards.add(stats.shards);
             }
         }
+        if (stats.getNodeMappings() != null) {
+            if (nodeMappings == null) {
+                nodeMappings = new NodeMappingStats();
+                nodeMappings.add(stats.getNodeMappings());
```

Comment:
Can't we just do what we do for shard stats above and go with:

```java
if (nodeMappings == null) {
    nodeMappings = stats.getNodeMappings();
} else {
   nodeMappings.add(stats.getNodeMappings());
}
```

?

Reply:
I believe shard stats is doing it very differently than the code above it, which actually creates a new object rather than assigning the reference of the added object (like what shard stats code is doing). So I decided to follow the above code, which seems better/safer to me. Feel free to tell me if you disagree and I can make it like shard stats

Reply:
Hah yea we have both versions. It doesn't really matter I guess, fine by me to go with the safe version here.